contract/src/contract.rs

```diff
@@ -129,6 +143,43 @@ pub fn migrate(deps: DepsMut, _env: Env, _msg: Empty) -> Result<Response, Contra
         return Err(ContractError::NoMigrationNeeded);
     }
 
+    const V_1_0_5: Version = Version::new(1, 0, 5);
+    // migrate from 1.0.4 to 1.0.5
+    if storage_version < V_1_0_5 {
+        #[cfg(test)]
+        {
+            // create the new DR config
+            let new_dr_config = OldDrConfig {
+                commit_timeout_in_blocks: INITIAL_COMMIT_TIMEOUT_IN_BLOCKS,
+                reveal_timeout_in_blocks: INITIAL_REVEAL_TIMEOUT_IN_BLOCKS,
+                backup_delay_in_blocks:   INITIAL_BACKUP_DELAY_IN_BLOCKS,
+            };
+            // Serialize the new data
+            let new_data = cosmwasm_std::to_json_vec(&new_dr_config)?;
+            // Store the new data
+            deps.storage.set(b"dr_config", &new_data);
+        }
+
+        // load the old DR config
+        let Some(old_data) = deps.storage.get(b"dr_config") else {
```

Comment:
Why don't we use the `DR_CONFIG` like we do in the initialiser?

Reply:
It has the wrong typing on it since that has the `DrConfig` struct that has the new field added. Doing it this way is the [recommended](https://cosmwasm.cosmos.network/core/entrypoints/migrate) way as well. Last time I didn't do it this way cause I didn't look at the docs :x.